gateway-js/src/index.ts

```diff
@@ -445,8 +405,25 @@ export class ApolloGateway implements GraphQLService {
     return result;
   }
 
-  private loadStatic() {
-    this.maybeWarnOnConflictingConfig();
+  private loadStatic(config: StaticGatewayConfig) {
+    const schemaConstructionOpts = isLocalConfig(config)
+      ? { serviceList: config.localServiceList }
+      : { csdl: config.csdl };
+
+    let schema: GraphQLSchema;
+    let composedSdl: string;
+    try {
+      ({ schema, composedSdl } = this.createSchema(schemaConstructionOpts));
+    } catch (e) {
+      throw Error(
+        "A valid schema couldn't be composed. The following errors were found:\n" +
```

Comment:
Is this error appropriate in the CSDL case? It wasn't there before this commit and the wording is a bit off for it.

(Though I do wonder why we need this try/catch anyway rather than have createSchema throw a useful error in the first place? What particular error is this trying to annotate?

Reply:
You're correct, `this.createSchema` flat out won't throw in the CSDL case so I can go back to splitting the cases via two `if`s.

This message precedes a list of graphql errors. We could just let that bubble up and not annotate it at all, or add this annotation over there instead?

Reply:
Yeah I guess I'm wondering if adding it more tightly would help. It's an error about "composing", it could be thrown in the compose function or right when that is called?

Reply:
This is a nice change and additionally gives us better control over how the list of composition errors is displayed (`GraphQLSchemaValidationError` forced multiple new lines between each error in the printing).
d66d683

Reply:
Cool. I don't think I understand the implications of switching from `GraphQLSchemaValidationError` to just a plain `Error` (eg nothing checks error types when catching, right?) but trust that you do.

Reply:
Yeah, this is a safe change (esp. since we were already just catching it and rethrowing as a plain `Error`).

There may have been some things to be careful of if it was public surface area / in use in other places, but not the case here.